src/hotspot/share/classfile/modules.cpp

```diff
@@ -649,17 +652,20 @@ void Modules::serialize_native_access_flags(SerializeClosure* soc) {
 }
 
 void Modules::dump_addmods_names() {
+  ResourceMark rm;
   const char* addmods_names = get_addmods_names_as_sorted_string();
   if (addmods_names != nullptr) {
     _archived_addmods_names = ArchiveBuilder::current()->ro_strdup(addmods_names);
   }
 }
 
 const char* Modules::get_addmods_names_as_sorted_string() {
```

Comment:
```suggestion
// Caller needs ResourceMark
const char* Modules::get_addmods_names_as_sorted_string() {
```

Reply:
I don't mind additional comment, but I prefer to keep assertion as well.